packages/stryker/src/transpiler/SourceMapper.ts

```diff
@@ -0,0 +1,181 @@
+import * as path from 'path';
+import { SourceMapConsumer, RawSourceMap } from 'source-map';
+import { File, FileKind, TextFile, Location } from 'stryker-api/core';
+import { Config } from 'stryker-api/config';
+import { base64Decode } from '../utils/objectUtils';
+
+const SOURCE_MAP_URL_REGEX = /\/\/\s*#\s*sourceMappingURL=(.*)/g;
+
+export interface MappedLocation {
+  fileName: string;
+  location: Location;
+}
+
+interface SourceMapInfo {
+  transpiledFile: TextFile;
+  sourceMapFileName: string;
+  sourceMap: SourceMapConsumer;
+}
+
+interface SourceMapInfoBySource {
+  [sourceFileName: string]: SourceMapInfo;
+}
+
+export class SourceMapError extends Error {
+  constructor(message: string) {
+    super(`${message}. Cannot analyse code coverage. Setting \`coverageAnalysis: "off"\` in your stryker.conf.js will prevent this error, but forces Stryker to run each test for each mutant.`);
+    Error.captureStackTrace(this, SourceMapError);
+    // TS recommendation: https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    Object.setPrototypeOf(this, SourceMapError.prototype);
+  }
+}
+
+export default abstract class SourceMapper {
+  abstract transpiledLocationFor(originalLocation: MappedLocation): MappedLocation;
+
+  static create(transpiledFiles: File[], config: Config): SourceMapper {
+    if (config.transpilers.length && config.coverageAnalysis !== 'off') {
+      return new TranspiledSourceMapper(transpiledFiles);
+    } else {
+      return new PassThroughSourceMapper();
+    }
+  }
+}
+
+export class TranspiledSourceMapper extends SourceMapper {
```

Comment:
I've refactored a bunch of code and made it more object-oriented by adding a `SourceMap` class (instead of a `SourceMapInfo` interface). Hope this is a bit clearer.